### ISSUE TITLE
Unable to compile on big sur using opencv and fmod third party packages

### ISSUE THREAD

**Issue by DarthNihilus1** [panda3d/panda3d#1062]
## Description

I expected it to compile but it fails. 
```
Version: 1.11.0
Platform: macosx-10.16-x86_64
Using Python 3.8 build located at /Library/Frameworks/Python.framework/Versions/3.8/Headers
Using compiler: GCC
Host OS: darwin
Host arch: x86_64
Target OS: darwin
Target arch: x86_64
System library search path:
  /usr/lib
  /usr/local/lib
  /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0
System include search path:
  /usr/local/include
  /Library/Developer/CommandLineTools/usr/include/c++/v1
  /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0/include
  /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
  /Library/Developer/CommandLineTools/usr/include
Using thirdparty directory: thirdparty/darwin-libs-a/
Couldn't find library libfftw3
Couldn't find header file fftw.h
WARNING: Could not locate thirdparty package fftw, excluding from build
Couldn't find the framework OpenGLES
Couldn't find library libGLESv1_CM
Couldn't find header file GLES/gl.h
WARNING: Could not locate thirdparty package gles, excluding from build
Couldn't find library libGLESv2
Couldn't find header file GLES2/gl2.h
WARNING: Could not locate thirdparty package gles2, excluding from build
Couldn't find library libEGL
Couldn't find header file EGL/egl.h
WARNING: Could not locate thirdparty package egl, excluding from build
WARNING: Could not locate pkg-config package gtk+-2.0, excluding from build
Writing built/pandac/extension_native_helpers.py
Removing file built/Frameworks/Cg.framework/Cg
Removing file built/Frameworks/Cg.framework/Resources
Removing file built/Frameworks/Cg.framework/Versions/Current
Removing file built/Frameworks/Cg.framework/Headers
Removing file built/lib/libopencv_highgui.2.4.dylib
install_name_tool -id @loader_path/../lib/libopencv_core.2.4.3.dylib built/lib/libopencv_core.2.4.3.dylib
error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: in swap_object_headers(): malformed load command 0 (cmdsize not a multiple of 4)
fatal error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: internal error: swap_object_headers() failed
Process exited with exit status 0 and signal code 1
Storing dependency cache.
Elapsed Time: 1 sec
The following command returned a non-zero value: install_name_tool -id @loader_path/../lib/libopencv_core.2.4.3.dylib built/lib/libopencv_core.2.4.3.dylib
Build terminated.

```
## Steps to Reproduce

The latest commit the source I use is from september 1st,. So checkout 6c8066bb619d263527e870c4a8148201ba9e3a3a then try to compile on big sur.
## Environment
 * Operating system: Mac OS 11.0, Macbook Air 2017
 * System architecture: 64 bit
 * Panda3D version: 1.11.0
 * Installation method: built from source
 * Python version (if using Python): python 3.8.6
 * Compiler (if using C++): clang 6.0 i believe

**Comment by romantolkachyov** [comment]
Same here

```
Version: 1.11.0
Platform: macosx-11.0-x86_64
Using Python 3.8 build located at /Library/Frameworks/Python.framework/Versions/3.8/Headers
Using compiler: GCC
Host OS: darwin
Host arch: x86_64
Target OS: darwin
Target arch: x86_64
System library search path:
  /usr/lib
  /usr/local/lib
  /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0
System include search path:
  /usr/local/include
  /Library/Developer/CommandLineTools/usr/include/c++/v1
  /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0/include
  /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
  /Library/Developer/CommandLineTools/usr/include
Using thirdparty directory: thirdparty/darwin-libs-a/
```

**Comment by rdb** [comment]
You haven't shared which command makepanda was doing when it gave this error (add `--verbose` for more info).  Please share more context.

**Comment by romantolkachyov** [comment]
Updated comment above

**Comment by DarthNihilus1** [comment]
Hope this helps:
`Version: 1.11.0
Platform: macosx-10.16-x86_64
Using Python 3.8 build located at /Library/Frameworks/Python.framework/Versions/3.8/Headers
Using compiler: GCC
Host OS: darwin
Host arch: x86_64
Target OS: darwin
Target arch: x86_64
System library search path:
  /usr/lib
  /usr/local/lib
  /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0
System include search path:
  /usr/local/include
  /Library/Developer/CommandLineTools/usr/include/c++/v1
  /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0/include
  /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
  /Library/Developer/CommandLineTools/usr/include
Using thirdparty directory: thirdparty/darwin-libs-a/
Couldn't find library libfftw3
Couldn't find header file fftw.h
WARNING: Could not locate thirdparty package fftw, excluding from build
Couldn't find the framework OpenGLES
Couldn't find library libGLESv1_CM
Couldn't find header file GLES/gl.h
WARNING: Could not locate thirdparty package gles, excluding from build
Couldn't find library libGLESv2
Couldn't find header file GLES2/gl2.h
WARNING: Could not locate thirdparty package gles2, excluding from build
Couldn't find library libEGL
Couldn't find header file EGL/egl.h
WARNING: Could not locate thirdparty package egl, excluding from build
WARNING: Could not locate pkg-config package gtk+-2.0, excluding from build
Writing built/pandac/extension_native_helpers.py
Removing file built/Frameworks/Cg.framework/Cg
Removing file built/Frameworks/Cg.framework/Resources
Removing file built/Frameworks/Cg.framework/Versions/Current
Removing file built/Frameworks/Cg.framework/Headers
Removing file built/lib/libopencv_highgui.2.4.dylib
install_name_tool -id @loader_path/../lib/libopencv_core.2.4.3.dylib built/lib/libopencv_core.2.4.3.dylib
error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: in swap_object_headers(): malformed load command 0 (cmdsize not a multiple of 4)
fatal error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: internal error: swap_object_headers() failed
Process exited with exit status 0 and signal code 1
Storing dependency cache.
Elapsed Time: 1 sec
The following command returned a non-zero value: install_name_tool -id @loader_path/../lib/libopencv_core.2.4.3.dylib built/lib/libopencv_core.2.4.3.dylib
Build terminated.
`

**Comment by rdb** [comment]
Strange.  When I do `otool -l built/lib/libopencv_core.2.4.3.dylib` I see for "Load command 0" at the top a "cmdsize" of 712, which really is a multiple of 4.  Do you see something different?

You could compile with `--no-opencv` to skip this library.

**Comment by romantolkachyov** [comment]
My output:

```
❯ otool -l built/lib/libopencv_core.2.4.3.dylib                                                                                    panda3d -> master ?
built/lib/libopencv_core.2.4.3.dylib:
Mach header
      magic cputype cpusubtype  caps    filetype ncmds sizeofcmds      flags
 0xfeedfacf 16777223          3  0x00           6    13       1920 0x00118085
Load command 0
      cmd LC_SEGMENT_64
  cmdsize 712
  segname __TEXT
   vmaddr 0x0000000000000000
   vmsize 0x00000000001e6000
  fileoff 0
 filesize 1990656
  maxprot 0x00000007
 initprot 0x00000005
   nsects 8
    flags 0x0
Section
  sectname __text
   segname __TEXT
      addr 0x0000000000000ab0
      size 0x0000000000188b76
    offset 2736
     align 2^4 (16)
    reloff 0
    nreloc 0
     flags 0x80000400
 reserved1 0
 reserved2 0
Section
  sectname __stubs
   segname __TEXT
      addr 0x0000000000189626
      size 0x000000000000082e
    offset 1611302
     align 2^1 (2)
    reloff 0
    nreloc 0
     flags 0x80000408
 reserved1 0 (index into indirect symbol table)
 reserved2 6 (size of stubs)
Section
  sectname __stub_helper
   segname __TEXT
      addr 0x0000000000189e54
      size 0x00000000000004fc
    offset 1613396
     align 2^2 (4)
    reloff 0
    nreloc 0
     flags 0x80000400
 reserved1 0
 reserved2 0
Section
  sectname __gcc_except_tab
   segname __TEXT
      addr 0x000000000018a350
      size 0x000000000001e8e0
    offset 1614672
     align 2^2 (4)
    reloff 0
    nreloc 0
     flags 0x00000000
 reserved1 0
 reserved2 0
Section
  sectname __const
   segname __TEXT
      addr 0x00000000001a8c30
      size 0x0000000000006ed1
    offset 1739824
     align 2^4 (16)
    reloff 0
    nreloc 0
     flags 0x00000000
 reserved1 0
 reserved2 0
Section
  sectname __cstring
   segname __TEXT
      addr 0x00000000001afb10
      size 0x000000000001bd41
    offset 1768208
     align 2^4 (16)
    reloff 0
    nreloc 0
     flags 0x00000002
 reserved1 0
 reserved2 0
Section
  sectname __unwind_info
   segname __TEXT
      addr 0x00000000001cb851
      size 0x0000000000002dc4
    offset 1882193
     align 2^0 (1)
    reloff 0
    nreloc 0
     flags 0x00000000
 reserved1 0
 reserved2 0
Section
  sectname __eh_frame
   segname __TEXT
      addr 0x00000000001ce618
      size 0x00000000000179e8
    offset 1893912
     align 2^3 (8)
    reloff 0
    nreloc 0
     flags 0x00000000
 reserved1 0
 reserved2 0
Load command 1
      cmd LC_SEGMENT_64
  cmdsize 712
  segname __DATA
   vmaddr 0x00000000001e6000
   vmsize 0x000000000000c000
  fileoff 1990656
 filesize 40960
  maxprot 0x00000007
 initprot 0x00000003
   nsects 8
    flags 0x0
Section
  sectname __got
   segname __DATA
      addr 0x00000000001e6000
      size 0x0000000000000118
    offset 1990656
     align 2^3 (8)
    reloff 0
    nreloc 0
     flags 0x00000006
 reserved1 349 (index into indirect symbol table)
 reserved2 0
Section
  sectname __nl_symbol_ptr
   segname __DATA
      addr 0x00000000001e6118
      size 0x0000000000000010
    offset 1990936
     align 2^3 (8)
    reloff 0
    nreloc 0
     flags 0x00000006
 reserved1 384 (index into indirect symbol table)
 reserved2 0
Section
  sectname __la_symbol_ptr
   segname __DATA
      addr 0x00000000001e6128
      size 0x0000000000000ae8
    offset 1990952
     align 2^3 (8)
    reloff 0
    nreloc 0
     flags 0x00000007
 reserved1 386 (index into indirect symbol table)
 reserved2 0
Section
  sectname __mod_init_func
   segname __DATA
      addr 0x00000000001e6c10
      size 0x0000000000000038
    offset 1993744
     align 2^3 (8)
    reloff 0
    nreloc 0
     flags 0x00000009
 reserved1 0
 reserved2 0
Section
  sectname __const
   segname __DATA
      addr 0x00000000001e6c50
      size 0x0000000000002078
    offset 1993808
     align 2^4 (16)
    reloff 0
    nreloc 0
     flags 0x00000000
 reserved1 0
 reserved2 0
Section
  sectname __data
   segname __DATA
      addr 0x00000000001e8cd0
      size 0x0000000000006ce8
    offset 2002128
     align 2^4 (16)
    reloff 0
    nreloc 0
     flags 0x00000000
 reserved1 0
 reserved2 0
Section
  sectname __common
   segname __DATA
      addr 0x00000000001ef9b8
      size 0x0000000000000068
    offset 0
     align 2^3 (8)
    reloff 0
    nreloc 0
     flags 0x00000001
 reserved1 0
 reserved2 0
Section
  sectname __bss
   segname __DATA
      addr 0x00000000001efa20
      size 0x0000000000001800
    offset 0
     align 2^4 (16)
    reloff 0
    nreloc 0
     flags 0x00000001
 reserved1 0
 reserved2 0
Load command 2
      cmd LC_SEGMENT_64
  cmdsize 72
  segname __LINKEDIT
   vmaddr 0x00000000001f2000
   vmsize 0x000000000003c000
  fileoff 2031616
 filesize 243012
  maxprot 0x00000007
 initprot 0x00000001
   nsects 0
    flags 0x0
Load command 3
          cmd LC_ID_DYLIB
      cmdsize 56
         name libopencv_core.2.4.dylib (offset 24)
   time stamp 1 Thu Jan  1 03:00:01 1970
      current version 2.4.3
compatibility version 2.4.0
Load command 4
            cmd LC_DYLD_INFO_ONLY
        cmdsize 48
     rebase_off 2031616
    rebase_size 328
       bind_off 2031944
      bind_size 728
  weak_bind_off 2032672
 weak_bind_size 23296
  lazy_bind_off 2055968
 lazy_bind_size 2936
     export_off 2058904
    export_size 49472
Load command 5
     cmd LC_SYMTAB
 cmdsize 24
  symoff 2111952
   nsyms 3276
  stroff 2167308
 strsize 107320
Load command 6
            cmd LC_DYSYMTAB
        cmdsize 80
      ilocalsym 0
      nlocalsym 1591
     iextdefsym 1591
     nextdefsym 1541
      iundefsym 3132
      nundefsym 144
         tocoff 0
           ntoc 0
      modtaboff 0
        nmodtab 0
   extrefsymoff 0
    nextrefsyms 0
 indirectsymoff 2164368
  nindirectsyms 735
      extreloff 0
        nextrel 0
      locreloff 0
        nlocrel 0
Load command 7
     cmd LC_UUID
 cmdsize 24
    uuid 42BDFFC2-0579-348C-BAC3-ED76558AC441
Load command 8
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.7
      sdk n/a
Load command 9
          cmd LC_LOAD_DYLIB
      cmdsize 48
         name /usr/lib/libz.1.dylib (offset 24)
   time stamp 2 Thu Jan  1 03:00:02 1970
      current version 1.2.5
compatibility version 1.0.0
Load command 10
          cmd LC_LOAD_DYLIB
      cmdsize 56
         name /usr/lib/libstdc++.6.dylib (offset 24)
   time stamp 2 Thu Jan  1 03:00:02 1970
      current version 52.0.0
compatibility version 7.0.0
Load command 11
          cmd LC_LOAD_DYLIB
      cmdsize 56
         name /usr/lib/libSystem.B.dylib (offset 24)
   time stamp 2 Thu Jan  1 03:00:02 1970
      current version 159.1.0
compatibility version 1.0.0
Load command 12
      cmd LC_FUNCTION_STARTS
  cmdsize 16
  dataoff 2108376
 datasize 3576
```

**Comment by romantolkachyov** [comment]
The next error I've got from `python makepanda/makepanda.py --everything --verbose --no-opencv`:

```
In file included from built/include/Eigen/Dense:1:
In file included from built/include/Eigen/Core:284:
built/include/Eigen/src/Core/Assign.h:57:41: warning: converting the enum constant to a boolean [-Wint-in-bool-context]
    MaySliceVectorize  = MightVectorize && DstHasDirectAccess
                                        ^
495 warnings generated.
[ 42%] Linking dynamic library built/lib/libp3fmod_audio.dylib
clang++ -dynamiclib -install_name @loader_path/../lib/libp3fmod_audio.dylib -compatibility_version 1.11 -current_version 1.11.0 -o built/lib/libp3fmod_audio.dylib -Lbuilt/lib -Lbuilt/tmp built/tmp/fmod_audio_fmod_audio_composite1.o built/lib/libpanda.dylib built/lib/libpandaexpress.dylib built/lib/libp3dtool.dylib built/lib/libp3dtoolconfig.dylib -headerpad_max_install_names -stdlib=libc++ -arch x86_64 built/lib/libfmodex.dylib -framework AppKit -ldl
ld: file not found: /usr/lib/libstdc++.6.dylib for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Process exited with exit status 0 and signal code 1
Storing dependency cache.
Elapsed Time: 19 min 29 sec
The following command returned a non-zero value: clang++ -dynamiclib -install_name @loader_path/../lib/libp3fmod_audio.dylib -compatibility_version 1.11 -current_version 1.11.0 -o built/lib/libp3fmod_audio.dylib -Lbuilt/lib -Lbuilt/tmp built/tmp/fmod_audio_fmod_audio_composite1.o built/lib/libpanda.dylib built/lib/libpandaexpress.dylib built/lib/libp3dtool.dylib built/lib/libp3dtoolconfig.dylib -headerpad_max_install_names -stdlib=libc++ -arch x86_64 built/lib/libfmodex.dylib -framework AppKit -ldl
Build terminated.
```

**Comment by rdb** [comment]
Try adding --no-fmod

**Comment by romantolkachyov** [comment]
Trying to do so.

By the way, I've tried clean Mac OS and nothing changed. Also, there is an error with python path (I've fixed it before faced opencv issue):

```
❯ python makepanda/makepanda.py --everything --verbose --no-opencv --no-fmod
Version: 1.11.0
Platform: macosx-11.0-x86_64
Storing dependency cache.
Elapsed Time: 0 sec
Could not locate Python installation at /Library/Frameworks/Python.framework/Versions/3.8
Build terminated.
```

There is no such directory if you are using pyenv. It can be fixed with symlink (might be helpful to someone):

```
sudo mkdir -p /Library/Frameworks/Python.framework/Versions/
sudo ln -s `pyenv prefix` /Library/Frameworks/Python.framework/Versions/3.8
```

I believe @DarthNihilus1 did something like that too.

**Comment by romantolkachyov** [comment]
Successfully built with `--no-opencv --no-fmod`. But can't find a package to install into virtualenv.

<details>
<summary>Build report</summary>
```
Makepanda Final Status Report
Makepanda: Compiler: GCC
Makepanda: Optimize: 3
Makepanda: Keep Pkg: PYTHON DIRECT GL TINYDISPLAY NVIDIACG EIGEN OPENAL VORBIS OPUS FFMPEG SWSCALE SWRESAMPLE ODE BULLET PANDAPHYSICS ZLIB PNG JPEG TIFF OPENEXR SQUISH FCOLLADA ASSIMP EGG FREETYPE HARFBUZZ VRPN OPENSSL ARTOOLKIT VISION MFC WX FLTK COCOA X11 PANDATOOL PVIEW DEPLOYTOOLS SKEL PANDAFX PANDAPARTICLESYSTEM CONTRIB SSE2 NEON
Makepanda: Omit Pkg: GLES GLES2 DX9 EGL FMODEX SPEEDTREE MAYA6 MAYA65 MAYA7 MAYA8 MAYA85 MAYA2008 MAYA2009 MAYA2010 MAYA2011 MAYA2012 MAYA2013 MAYA20135 MAYA2014 MAYA2015 MAYA2016 MAYA20165 MAYA2017 MAYA2018 MAYA2019 MAYA2020 MAX6 MAX7 MAX8 MAX9 MAX2009 MAX2010 MAX2011 MAX2012 MAX2013 MAX2014 FFTW OPENCV DIRECTCAM GTK2
Makepanda: Don't generate API reference manual
Makepanda: Version ID: 1.11.0
Makepanda: I cannot locate SDK for MAYA6
Makepanda: I have automatically added this command-line option: --no-maya6
Makepanda: I cannot locate SDK for MAYA65
Makepanda: I have automatically added this command-line option: --no-maya65
Makepanda: I cannot locate SDK for MAYA7
Makepanda: I have automatically added this command-line option: --no-maya7
Makepanda: I cannot locate SDK for MAYA8
Makepanda: I have automatically added this command-line option: --no-maya8
Makepanda: I cannot locate SDK for MAYA85
Makepanda: I have automatically added this command-line option: --no-maya85
Makepanda: I cannot locate SDK for MAYA2008
Makepanda: I have automatically added this command-line option: --no-maya2008
Makepanda: I cannot locate SDK for MAYA2009
Makepanda: I have automatically added this command-line option: --no-maya2009
Makepanda: I cannot locate SDK for MAYA2010
Makepanda: I have automatically added this command-line option: --no-maya2010
Makepanda: I cannot locate SDK for MAYA2011
Makepanda: I have automatically added this command-line option: --no-maya2011
Makepanda: I cannot locate SDK for MAYA2012
Makepanda: I have automatically added this command-line option: --no-maya2012
Makepanda: I cannot locate SDK for MAYA2013
Makepanda: I have automatically added this command-line option: --no-maya2013
Makepanda: I cannot locate SDK for MAYA20135
Makepanda: I have automatically added this command-line option: --no-maya20135
Makepanda: I cannot locate SDK for MAYA2014
Makepanda: I have automatically added this command-line option: --no-maya2014
Makepanda: I cannot locate SDK for MAYA2015
Makepanda: I have automatically added this command-line option: --no-maya2015
Makepanda: I cannot locate SDK for MAYA2016
Makepanda: I have automatically added this command-line option: --no-maya2016
Makepanda: I cannot locate SDK for MAYA20165
Makepanda: I have automatically added this command-line option: --no-maya20165
Makepanda: I cannot locate SDK for MAYA2017
Makepanda: I have automatically added this command-line option: --no-maya2017
Makepanda: I cannot locate SDK for MAYA2018
Makepanda: I have automatically added this command-line option: --no-maya2018
Makepanda: I cannot locate SDK for MAYA2019
Makepanda: I have automatically added this command-line option: --no-maya2019
Makepanda: I cannot locate SDK for MAYA2020
Makepanda: I have automatically added this command-line option: --no-maya2020
Makepanda: I cannot locate SDK for SpeedTree
Makepanda: I have automatically added this command-line option: --no-speedtree
Makepanda: Elapsed Time: 5 sec
-------------------------------------------------------------------

Build successfully finished, elapsed time: 5 sec
```
</details>

**Comment by rdb** [comment]
You need to add the --wheel flag to get a .whl to install, or --installer to get a .dmg file.

**Comment by romantolkachyov** [comment]
Had to rebuild python with `--enable-framework` to make `--wheel` work. And looks like it was successfully installed. Cool! Thank you @rdb! 

Hate Apple. They've broke everything another time.

What I'm missing without opencv and fmod?

P.S> I've also had to solve macos security protection: it won't run Cg.Framework because it is unsigned. You need to go to the Security and Privacy settings, wait for "allow" button to appear while build script running, click on allow and restart build (it requires me 2 times to repeat before it starts working and macos allows me to choose "open" when asking me about Cg.Framework execution)

**Comment by rdb** [comment]
You're not missing much.  OpenCV is just used for the WebcamVideo class, and FMOD is a proprietary sound framework (but you can just use OpenAL instead).

Thanks for the notice about Cg.  Fortunately we will be removing the Cg framework in 1.11.0.

**Comment by DarthNihilus1** [comment]
> Trying to do so.
> 
> By the way, I've tried clean Mac OS and nothing changed. Also, there is an error with python path (I've fixed it before faced opencv issue):
> 
> ```
> ❯ python makepanda/makepanda.py --everything --verbose --no-opencv --no-fmod
> Version: 1.11.0
> Platform: macosx-11.0-x86_64
> Storing dependency cache.
> Elapsed Time: 0 sec
> Could not locate Python installation at /Library/Frameworks/Python.framework/Versions/3.8
> Build terminated.
> ```
> 
> There is no such directory if you are using pyenv. It can be fixed with symlink (might be helpful to someone):
> 
> ```
> sudo mkdir -p /Library/Frameworks/Python.framework/Versions/
> sudo ln -s `pyenv prefix` /Library/Frameworks/Python.framework/Versions/3.8
> ```
> 
> I believe @DarthNihilus1 did something like that too.

What i did was i used the official python3 installer from python.org instead of brew

**Comment by rdb** [comment]
I just created a 1.10.8 thirdparty package [here](https://www.panda3d.org/download/panda3d-1.10.8/panda3d-1.10.8-tools-mac.tar.gz) containing a recompiled OpenCV which should fix the issue with OpenCV.

As for FMOD, could someone let me know whether libstdc++.6.dylib exists at all on their Big Sur system, even in other locations?  If Apple has entirely removed that file, we may need to just disable FMOD entirely when building on/for 11.0.

**Comment by treamology** [comment]
From [macOS Big Sur 11.0.1 Release Notes](https://developer.apple.com/documentation/macos-release-notes/macos-big-sur-11_0_1-release-notes):

> - New in macOS Big Sur 11.0.1, the system ships with a built-in dynamic linker cache of all system-provided libraries. As part of this change, copies of dynamic libraries are no longer present on the filesystem. Code that attempts to check for dynamic library presence by looking for a file at a path or enumerating a directory will fail. Instead, check for library presence by attempting to dlopen() the path, which will correctly check for the library in the cache. (62986286)

Also, why is `libstdc++` being looked for when `-stdlib=libc++` is specified in the build command?

**Comment by rdb** [comment]
The FMOD library is a binary thirdparty, which I presume is still compiled against libstdc++.  Nothing we can do about that (except update to the latest FMOD—that's being worked on in #1032).

**Comment by paustian** [comment]
> I just created a 1.10.8 thirdparty package [here](https://www.panda3d.org/download/panda3d-1.10.8/panda3d-1.10.8-tools-mac.tar.gz) containing a recompiled OpenCV which should fix the issue with OpenCV.
> 
> As for FMOD, could someone let me know whether libstdc++.6.dylib exists at all on their Big Sur system, even in other locations? If Apple has entirely removed that file, we may need to just disable FMOD entirely when building on/for 11.0.

I can confirm the build compiles now on Big Sur with the opencv library

**Comment by rdb** [comment]
Could someone with Big Sur try to find out for me whether libstdc++.6.dylib is present in the new dynamic linker cache, or list the libraries in there?  I think this should be possible with `dyld_shared_cache_util -list`.

**Comment by paustian** [comment]
I had to build dyld_shared_cache_util but after it was built I was able to run it. I issued this command
`/dyld_shared_cache_util -list | grep libstdc`

And yep, its there. (Here is the output)

/usr/lib/libstdc++.6.dylib

I can post the entire giant list if you think it would be useful.

**Comment by rdb** [comment]
The problem is apparently that libstdc++.6.dylib is a symlink to libstdc++.6.0.9.dylib, which was removed starting with the macOS 10.14 SDK.  Not sure why it hasn't been a problem until now, but it might have to do with changed linker behaviour.

In any case, I've made the following changes to support Big Sur / "Apple Silicon":
* FMOD is automatically disabled by the build system except when targeting the 10.9 SDK.
* I've rebuilt the thirdparty packages to contain arm64 binaries [here](https://www.panda3d.org/download/panda3d-1.10.8/panda3d-1.10.8-tools-mac.tar.gz).
* It's now possible to target arm64 in makepanda using the --universal flag or by specifying multiple --arch flags.  It will still target 10.9 as minimum supported version when building a universal build, but using the 11.0 or 11.1 SDK.
